lib/rst/interp_float/point2d.c

```diff
@@ -193,3 +193,83 @@ int IL_check_at_points_2d(struct interp_params *params,
 
     return 1;
 }
+
+int IL_check_at_points_2d_cvdev(struct interp_params *params,
+                                struct quaddata *data, /*!< current region */
+                                double *b, /*!< solution of linear equations */
+                                double *ertot, /*!< total error */
+                                double zmin,   /*!< min z-value */
+                                double dnorm, struct triple *target_point)
+{
+    int n_points = data->n_points;        /* number of points */
+    struct triple *points = data->points; /* points for interpolation */
+    double west = data->x_orig;
+    double south = data->y_orig;
+    double /* rfsta2, errmax, */ h, xx, yy, r2, hz, zz, err, xmm, ymm, r;
+    int /* n1, */ m;
+
+    if (params->cv) { /* one point is skipped for cross-validation*/
+        n_points -= 1;
+    }
+
+    h = b[0];
+    for (m = 1; m <= n_points; m++) {
+        xx = target_point->x - points[m - 1].x;
+        yy = target_point->y - points[m - 1].y;
+        r2 = yy * yy + xx * xx;
+        if (r2 != 0.) {
+            /* rfsta2 = fstar2 * r2; */
+            r = r2;
+            h = h + b[m] * params->interp(r, params->fi);
+        }
+    }
+    /* modified by helena january 1997 - normalization of z was
+        removed from segm2d.c and interp2d.c
+        hz = (h * dnorm) + zmin;
+        zz = (points[mm - 1].z * dnorm) + zmin;
+    */
+    hz = h + zmin;
+    zz = target_point->z + zmin;
+    err = hz - zz;
+    xmm = target_point->x * dnorm + params->x_orig + west;
+    ymm = target_point->y * dnorm + params->y_orig + south;
+    (*ertot) += err * err;
+
+    /* take the values out*/
+    target_point->x = xmm;
+    target_point->y = ymm;
+    target_point->z = err;
+
+    return 1;
+}
+
+int IL_write_point_2d(struct triple point, double err)
```

Comment:
I added documentation in the new commit. Since this function only needs to write out values instead of updating or returning them, pass-by-value is used here, as `IL_check_at_points_2d` did before this PR.